src/Features/Core/Portable/RemoveAsyncModifier/AbstractRemoveAsyncModifierCodeFixProvider.cs

```diff
@@ -0,0 +1,236 @@
+            foreach (TReturnStatementSyntax returnSyntax in returns)
+            {
+                // Make sure we're not changing returns in nested local functions or lambdas
+                var containingMethod = returnSyntax.FirstAncestorOrSelf<SyntaxNode>(IsAsyncSupportingFunctionSyntax);
+                if (containingMethod != node)
+                {
+                    continue;
+                }
+
+                var returnExpression = generator.SyntaxFacts.GetExpressionOfReturnStatement(returnSyntax);
+                if (returnExpression is null)
+                {
+                    // Convert return; into return Task.CompletedTask;
+                    var returnTaskCompletedTask = GetReturnTaskCompletedTaskStatement(knownTypes, generator);
+                    editor.ReplaceNode(returnSyntax, returnTaskCompletedTask);
+                }
+                else
+                {
+                    // Convert return <expr>; into return Task.FromResult(<expr>);
+                    WrapExpressionWithTaskFromResult(returnExpression, editor, knownTypes);
+                }
+            }
+        }
+
+        private static void AppendTaskCompletedTaskReturn(SyntaxNode lastNode, SyntaxEditor editor, KnownTypes knownTypes)
+        {
+            var generator = editor.Generator;
+            var returnTaskCompletedTask = GetReturnTaskCompletedTaskStatement(knownTypes, generator);
+
+            editor.InsertAfter(lastNode, returnTaskCompletedTask);
+        }
+
+        private static SyntaxNode GetReturnTaskCompletedTaskStatement(KnownTypes knownTypes, SyntaxGenerator generator)
+        {
+            var taskCompletedTaskInvocation = GetTaskCompletedTaskInvocation(knownTypes, generator);
+            var statement = generator.ReturnStatement(taskCompletedTaskInvocation);
+            return statement;
+        }
+
+        private static TExpressionSyntax GetTaskCompletedTaskInvocation(KnownTypes knownTypes, SyntaxGenerator generator)
+        {
+            var taskTypeExpression = TypeExpressionForStaticMemberAccess(generator, knownTypes._taskType);
+            return (TExpressionSyntax)generator.MemberAccessExpression(taskTypeExpression, nameof(Task.CompletedTask));
+        }
+
+        private static void WrapExpressionWithTaskFromResult(SyntaxNode expression, SyntaxEditor editor, KnownTypes knownTypes)
+        {
+            var generator = editor.Generator;
+
+            var taskTypeExpression = TypeExpressionForStaticMemberAccess(generator, knownTypes._taskType);
+            var taskFromResult = generator.MemberAccessExpression(taskTypeExpression, nameof(Task.FromResult));
+            var invocation = generator.InvocationExpression(taskFromResult, expression.NormalizeWhitespace());
+            editor.ReplaceNode(expression, invocation);
+        }
+
+        // Workaround for https://github.com/dotnet/roslyn/issues/43950
```

Comment:
we should just fix that issue.  i esp. don't like the copy/pasting as oposed to having this nastyness in one centralized helper that things can call.